src/main/java/bit/javaoop/AbstractEmployee.java

```diff
@@ -1,8 +1,8 @@
 package bit.javaoop;
 
 public abstract class AbstractEmployee {
-    private String name;
-    private double salary;
+    protected String name;
```

Comment:
Zmusiłeś mnie do poczytania o użyciu modyfikatora protected, bo wydaje się, że użycie go w tym wypadku to idealny use-case. 
Wszędzie, gdzie bym nie spojrzał, są dwa stanowiska: 
+ używać protected kiedy się da
+ używać private kiedy się da

Uprościłem może je trochę, ale tak to mniej więcej wygląda.

Jedni argumentują, że lepsza hermetyzacja, gdy używamy private i większa kontrola kodu.
Drudzy, że protected, bo nigdy nie wiemy kiedy będzie trzeba nadpisać wartość tej zmiennej.

No i ogólnie wniosek nasuwa mi się taki, private używamy, gdy nasz projekt nie ma w zamierzeniu być używany kiedyś przez innych, jako zewnętrzna biblioteka.
Protected, gdy ma.
Bo gdy użyjemy protected, ktoś używający naszego kodu jako biblioteki, może bezproblemowo dziedziczyć z naszej klaski i zmienić sobie wartość danego pola, bez konieczności przekopiowywania całego kodu i przepisywania klasy tak, by mu się zgadzało.

Przykład:
```
class Animal {
 private int speed = 60;
 private void run() { ... }
}
```
Każde zwierzę ma zakodowaną prędkość równą 60. Teraz jeśli ktoś chciałby tam dać coś innego, nie jest w stanie tego zrobić, ze względu na brak dostępu. I teraz jeśli klasa Animal to nie nasz kod tylko zewnętrzna biblioteka to musimy przekopiować cały ten fragment kodu jako nasza klasa i zmienić modyfikator dostępu - słabo.

Dlatego reasumując, private do własnych projektów, nie zamierzonych do używania przez nikogo, protected gdy piszemy bibliotekę - najczęściej jakąś open-sourcową.

Odnosząc się na koniec do naszego przypadku - czy może być protected? Myślę, że tak. Dodatkowo fajnie widać, że dziedziczenie zachodzi ;)